test/testdata/namer/fuzz_shared_singletons.rb

```diff
@@ -2,4 +2,4 @@
 # disable-fast-path: true
 A=class A; end # error: Redefining constant
 
-class A; end # error: Redefining constant
+class A; end
```

Comment:
The reason this error changed is because we've already created a `class A` and we can reopen that class within the same file, so this simply finds the already-created `class A` from above.

Reply:
after processing this file, what is `A`?
is it class or symbol?
Which one "wins"?

Reply:
In this case, the constant `A` wins, because the first `class A; end` introduces `A`, then the constant `A` gets initialized and renames the existing `A` to `A$1`, then the second `class A; end` gets treated as reopening `A$1` but doesn't do any mangling. Should we prefer a different semantics?

Reply:
> Should we prefer a different semantics?

There are downsides & upsides to both.
I think this PR introduces a subtle change and thus it's worth giving it more tests and writing a lot of text for it.